src/Compiler/Utilities/Cancellable.fs

```diff
@@ -68,8 +68,9 @@ module Cancellable =
             try
                 use _ = Cancellable.UsingToken(ct)
                 oper ct
-            with :? OperationCanceledException as e ->
-                ValueOrCancelled.Cancelled(OperationCanceledException e.CancellationToken)
+            with
+            | :? OperationCanceledException as e when e.CancellationToken = ct -> ValueOrCancelled.Cancelled e
```

Comment:
```suggestion
            | :? OperationCanceledException as e when ct.IsCancellationRequested -> ValueOrCancelled.Cancelled e
```

I think that equality might not hold if a call went trough creation of linked tokens.

Reply:
Example:

This "ct" is actually a linked token created in an upper layer (e.g. Roslyn or VS).
When the top token gets cancelled, the exception carries that top token - despite not being equal, our token still acts as IsCancellationRequested

Reply:
Yeah, I've also figured it out now. I wonder if there's much sense in this check at all, since two independent tokens may get cancelled. Though, we could still catch cases where another token throws an exception while our token is not requested to cancel yet.

Reply:
> In the case of a different token getting cancelled, and ours not - this should be IMO treated as a programming error (i.e. the stacktrace below our program has introduced a different token we were not aware of).

Yes, this is what I wanted to assert here 🙂